charts/internal/seed-controlplane/charts/csi-driver-manila-controller/templates/deployment-controller.yaml

```diff
@@ -0,0 +1,286 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: csi-driver-manila-controller
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: csi
+    role: driver-manila-controller
+    high-availability-config.resources.gardener.cloud/type: controller
+spec:
+  replicas: {{ .Values.replicas }}
+  revisionHistoryLimit: 1
+  selector:
+    matchLabels:
+      app: csi
+      role: driver-manila-controller
+  strategy:
+    rollingUpdate:
+      maxSurge: 25%
+      maxUnavailable: 25%
+    type: RollingUpdate
+  template:
+    metadata:
+{{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8 }}
+{{- end }}
+      creationTimestamp: null
+      labels:
+        app: csi
+        role: driver-manila-controller
+        gardener.cloud/role: controlplane
+        networking.gardener.cloud/to-dns: allowed
+        networking.gardener.cloud/to-public-networks: allowed
+        networking.gardener.cloud/to-private-networks: allowed
+        networking.resources.gardener.cloud/to-kube-apiserver-tcp-443: allowed
+    spec:
+      automountServiceAccountToken: false
+      priorityClassName: gardener-system-300
+      containers:
+        {{- range .Values.shareProtocols }}
```

Comment:
I am a bit conflicted. For now we only care about NFS but even if we did have more protocols that we want to support, I am not sure if I would prefer to run them in a "mega"-pod rather than its own separate deployments.

Reply:
dropped the range over share protocols